core/templates/dev/head/player/PlayerServices.js

```diff
@@ -197,31 +196,17 @@ oppia.factory('oppiaPlayerService', [
 
     _stopwatch.resetStopwatch();
 
-    var newStateData = _exploration.states[newStateName];
     $rootScope.$broadcast('playerStateChange');
     successCallback(
       newStateName, refreshInteraction, newFeedbackHtml, newQuestionHtml,
       newParams);
   };
 
-  var _registerMaybeLeaveEvent = function(stateName) {
-    var maybeLeaveExplorationUrl = (
-      '/explorehandler/exploration_maybe_leave_event/' + _explorationId);
-    $http.post(maybeLeaveExplorationUrl, {
-      client_time_spent_in_secs: _stopwatch.getTimeInSecs(),
-      params: learnerParamsService.getAllParams(),
-      session_id: sessionId,
-      state_name: stateName,
-      version: version
-    });
-  };
-
-  // This should only be called when _exploration is non-null.
+  // This should only be called when 'exploration' is non-null.
   var _loadInitialState = function(successCallback) {
-    var initStateName = _exploration.init_state_name;
     var initStateData = stateTransitionService.getInitStateData(
-      _exploration.param_specs, _exploration.param_changes,
-      _exploration.states[initStateName]);
+      exploration.paramSpecs, exploration.paramChanges,
```

Comment:
OK -- I got rid of the stateTransitionService file anyway, since it seemed a bit neither-here-nor-there. I moved part of it into the expressions/ folder and the other part into PlayerServices (which was its only consumer).